app/models/concerns/cache_keys.rb

```diff
@@ -4,29 +4,41 @@ module CacheKeys
   include CacheKeysHelper
   include Events::Types
 
+  included do
+    class_attribute :cacheable_models
+    self.cacheable_models = [Label, Inbox, Team]
+  end
+
   def cache_keys
-    {
-      label: fetch_value_for_key(id, Label.name.underscore),
-      inbox: fetch_value_for_key(id, Inbox.name.underscore),
-      team: fetch_value_for_key(id, Team.name.underscore)
-    }
+    keys = {}
+    self.class.cacheable_models.each do |model|
+      keys[model.name.underscore.to_sym] = fetch_value_for_key(id, model.name.underscore)
+    end
+
+    keys
   end
 
   def invalidate_cache_key_for(key)
     prefixed_cache_key = get_prefixed_cache_key(id, key)
```

Comment:
```
 def get_prefixed_cache_key(account_id, key)
    "idb-cache-key-account-#{account_id}-#{key}"
  end
```

@scmmishra  We are expecting account_id as the argument for the cache keys right ?
but it seems like we are passing get_prefixed_cache_key(id, key)  -> id here ? which is the object id ? 

could that be the reason for original issue ?

Reply:
Let me explore this

Reply:
@sojan-official just confirmed, the ID passed here is the ID for the account

Reply:
@scmmishra  Can we add a spec to ensure that it's the correct cache that gets cleared?

Reply:
Sure, I'll do that

Reply:
@sojan-official done